test/test_common.c

```diff
@@ -219,44 +273,65 @@ sport_new (const char *optarg, struct prog *prog)
     sport->packs_in = NULL;
     sport->fd = -1;
     char *const addr = strdup(optarg);
+    char ip[INET6_ADDRSTRLEN];
 #if __linux__
     char *if_name;
     if_name = strrchr(addr, ',');
     if (if_name)
     {
         strncpy(sport->if_name, if_name + 1, sizeof(sport->if_name) - 1);
-        sport->if_name[ sizeof(sport->if_name) - 1 ] = '\0';
+        sport->if_name[sizeof(sport->if_name) - 1] = '\0';
         *if_name = '\0';
     }
     else
         sport->if_name[0] = '\0';
 #endif
     char *port = strrchr(addr, ':');
-    if (!port)
-        goto err;
-    *port = '\0';
-    ++port;
-    if ((uintptr_t) port - (uintptr_t) addr > sizeof(sport->host))
-        goto err;
-    memcpy(sport->host, addr, port - addr);
+    if (!port)/*IpAdress wasn't specified by the user*/
+    {
+        if (get_Ip_from_DNS(prog->prog_hostname, ip, ipv6, addr, INET6_ADDRSTRLEN) == 1)
+        {
+            LSQ_ERROR("Couldn't resolve the hostname.\n");
+            goto err;
+        }
+        port = strdup(optarg);
+
+        memcpy(sport->host, ip, sizeof(ip));
+    }
+    else /*The user specified an ipaddress*/
```

Comment:
IP address or a hostname.

Reply:
Shouldn't it it be and a hostname since the if statement would be executed if only a hostname is specified?

Reply:
It could be either an IP address (`-s 1.2.3.4:443`) or a hostname (`-s example.com:443`).  If both `inet_pton()` calls fail, it must be a hostname.

Reply:
Oh so you want the hostname to also be specifyable in the -s flag? Since it is a required flag I always took the hostname from the -H flag.

Reply:
Yes: if the host part of the address (`-s` option) is not specified, use the SNI (`-H` option) as fallback.

Reply:
I don't think anyone would enter the hostname twice + port instead of hostname + only port but if you want I can add that